### ISSUE TITLE
Problem with building on Ubuntu 14.04 LTS 32-Bit

### ISSUE THREAD

**Issue by manu-m** [netty/netty#4694]
Hi!

I would like to use netty for communication with a native application via Unix Domain Sockets and I tried to use transport-native-epoll module but only artifacts for 64-Bit systems are officially provided at the moment. 

So I tried to build netty-tcnative and transport-epoll-module on my own, but I got an error while building tcnative c sources:

>[INFO] src/ssl.c: In function 'Java_org_apache_tomcat_jni_SSL_newMemBIO':
>[INFO] src/ssl.c:1070:16: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
>[INFO]          return (jlong) NULL;
>[INFO]                 ^
>[INFO] cc1: all warnings being treated as errors
>[INFO] make: *** [src/ssl.lo] Fehler 1
>[INFO] rc: 2

How to fix this error correctly? I added an additional cast to `intptr_t` before casting to `jlong`:
```c
return (jlong)(intptr_t) NULL
```

After this change I was able to execute `mvn clean install` without any problems for *tcnative* (but I'm not sure if this fix is correct because I'm not familiar with JNI). 

Then I tried to build transport-native-epoll. I got similar errors (in several files) to the one listed above. For example, the errors for io_netty_channel_unix_Socket.c are:

>[INFO] libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./src -O3 -Werror -I/opt/jdk1.8.0_65//include -I/opt/jdk1.8.0_65//include/linux -c src/io_netty_channel_unix_Socket.c  -fPIC -DPIC -o src/.libs/io_netty_channel_unix_Socket.o
>[INFO] src/io_netty_channel_unix_Socket.c: In function 'Java_io_netty_channel_unix_Socket_sendToAddress':
>[INFO] src/io_netty_channel_unix_Socket.c:541:29: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
>[INFO]      return _sendTo(env, fd, (void*) memoryAddress, pos, limit, address, scopeId, port);
>[INFO]                              ^
>[INFO] src/io_netty_channel_unix_Socket.c: In function 'Java_io_netty_channel_unix_Socket_sendToAddresses':
>[INFO] src/io_netty_channel_unix_Socket.c:554:17: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
>[INFO]      m.msg_iov = (struct iovec*) memoryAddress;
>[INFO]                  ^
>[INFO] src/io_netty_channel_unix_Socket.c: In function 'Java_io_netty_channel_unix_Socket_recvFromAddress':
>[INFO] src/io_netty_channel_unix_Socket.c:576:31: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
>[INFO]      return _recvFrom(env, fd, (void*) address, pos, limit);
>[INFO]                                ^
>[INFO] cc1: all warnings being treated as errors
>[INFO] make: *** [src/io_netty_channel_unix_Socket.lo] Fehler 1
>[INFO] rc: 2

I fixed these errors in the same way (as mentioned for tcnative) by adding an additional cast to `intptr_t`:
```c
return _sendTo(env, fd, (void*)(intptr_t) memoryAddress, pos, limit, address, scopeId, port);
```

After fixing all errors in this way I was able to build transport-native-epoll, but the unit tests that relate to SSL failed. Perhaps because my fix (of ssl.c of tcnative) was not correct?

Is there a chance to get transport-native-epoll running on 32-Bit systems? If so, how do I have to fix the problems mentioned above correctly? 

Thank you very much!

**Comment by normanmaurer** [comment]
Looking into it.

**Comment by normanmaurer** [comment]
Fixed by https://github.com/netty/netty/pull/4712 and https://github.com/netty/netty-tcnative/pull/83

**Comment by manu-m** [comment]
Did you try run it on a 32-Bit machine? Building now works, but some unit tests fail as it was the case after I fixed the problems in a very similar way.

All errors are in the context of SSL. If I don't use SSL for communication via Unix Domain Sockets, can I ignore for the short time?

I attached the reports:
[io.netty.channel.epoll.EpollDomainSocketSslEchoTest.txt](https://github.com/netty/netty/files/91897/io.netty.channel.epoll.EpollDomainSocketSslEchoTest.txt)
[io.netty.channel.epoll.EpollSocketSslEchoTest.txt](https://github.com/netty/netty/files/91894/io.netty.channel.epoll.EpollSocketSslEchoTest.txt)

**Comment by normanmaurer** [comment]
@manu-m yes ... I used oracle jdk 8 and it worked

**Comment by manu-m** [comment]
Which Linux Distro and which version?

**Comment by normanmaurer** [comment]
@manu-m exactly what you used... 'Ubuntu 14.04 LTS 32-Bit' . I set it up to reproduce .

**Comment by manu-m** [comment]
Hmm... strange. How did you build tc-native and netty and which branch do I have to checkout? Current netty 4.0 branch has not a dependency to tcnative-fork11 snapshot. Why? 

I checked out tcnative 1.1.33 branch and built it with `mvn clean install`. Then I checked out netty 4.0 branch, changed tcnative dependency to version 1.1.33.Fork11-SNAPSHOT and built transport-native-epoll also with `mvn clean install`. Some tests are failing.
